solana/solana-ibc/programs/solana-ibc/src/transfer/impls.rs

```diff
@@ -39,12 +39,9 @@ fn get_escrow_account(
     channel_id: &ChannelId,
     denom: &str,
 ) -> Pubkey {
-    let (head, tail) = stdx::split_at::<32>(denom.as_bytes())
-        .map(|(head, tail)| (&head[..], tail))
-        .unwrap_or((denom.as_bytes(), &b""[..]));
-    let seeds = [port_id.as_bytes(), channel_id.as_bytes(), head, tail];
-    let seeds = if tail.is_empty() { &seeds[..3] } else { &seeds[..] };
-    Pubkey::find_program_address(seeds, &crate::ID).0
+    let denom = hash::hash(denom.as_bytes()).to_bytes();
+    let seeds = [port_id.as_bytes(), channel_id.as_bytes(), &denom];
```

Comment:
Let’s do CryptoHash here as well so we consistently use this type
everywhere.

```suggestion
    let denom = lib::hash::CryptoHash::digest(denom.as_bytes());
    let seeds = [port_id.as_bytes(), channel_id.as_bytes(), denom.as_slice()];
```

Reply:
yes makes sense.